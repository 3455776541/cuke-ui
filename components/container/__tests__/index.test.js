import React from "react";
import assert from "power-assert";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import Container from "../index";

describe("<Container/>", () => {
	it("should render a <Container/> components", () => {
		const wrapper = render(
			<div>
				<Container>默认1200px 居中</Container>

				<Container width={200}>200px 居中</Container>

				<Container width={500} center>
					内容垂直水平居中
				</Container>
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should find cuke-container classnames", () => {
		const wrapper = render(
			<div>
				<Container>默认1200px 居中</Container>

				<Container width={500} center>
					内容垂直水平居中
				</Container>
			</div>
		);
		assert(wrapper.find(".cuke-container").length >= 1);
		assert(wrapper.find(".cuke-container-centered").length === 1);
	});
});
