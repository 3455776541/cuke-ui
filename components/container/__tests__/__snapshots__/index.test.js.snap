// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container/> should render a <Container/> components 1`] = `
<div>
  <div
    class="cuke-container"
    style="width:1200px"
  >
    <div
      class="cuke-container-wrap"
    >
      默认1200px 居中
    </div>
  </div>
  <div
    class="cuke-container"
    style="width:200px"
  >
    <div
      class="cuke-container-wrap"
    >
      200px 居中
    </div>
  </div>
  <div
    class="cuke-container cuke-container-centered"
    style="width:500px"
  >
    <div
      class="cuke-container-wrap"
    >
      内容垂直水平居中
    </div>
  </div>
</div>
`;
